src/algebraic_geometry/projective_spectrum/structure_sheaf.lean

```diff
@@ -0,0 +1,233 @@
+/-
+Copyright (c) 2022 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+import algebraic_geometry.projective_spectrum.topology
+import topology.sheaves.local_predicate
+import ring_theory.graded_algebra.homogeneous_localization
+import algebraic_geometry.locally_ringed_space
+
+/-!
+# The structure sheaf on `projective_spectrum 𝒜`.
+
+In `src/algebraic_geometry/topology.lean`, we have given a topology on `projective_spectrum 𝒜`; in
+this file we will construct a sheaf on `projective_spectrum 𝒜`.
+
+## Notation
+- `R` is a commutative semiring;
+- `A` is a commutative ring and an `R`-algebra;
+- `𝒜 : ℕ → submodule R A` is the grading of `A`;
+- `U` is opposite object of some open subset of `projective_spectrum.Top`.
+
+## Main definitions and results
+* `projective_spectrum.Top`: the topological space of `projective_spectrum 𝒜` endowed with the
```

Comment:
Shouldn't this be done in the topology file?

Reply:
done